### PR TITLE
Compound parameter editor improvements : Fix #351

### LINKED ISSUE #351: [CompoundParameterEditor] Improvement
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of the COMET-WEB
- [x] I have searched [open](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/issues) and [closed](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description

The CompoundParameterEditor can have some improvement.

- [ ] IsOnEditMode should be related to the Component, not to the VM. This leads to weird behavior with the 3DViewer
- [ ] The Input mode is not set on contained component
- [ ] Inside the ParameterEditor, a compoundParameter does not react to the change of the Switch 
### Logs

  - [ ] There are no logs regarding the issue
  - [ ] I have attached a log file or included the relevant section in this issue  
  
### System Configuration

- COMET version:
  - [ ] COMET-WEB: 
  - [ ] Other: 
- Web Services:
  - [ ] COMET Web Services version:

### PR DESCRIPTION
**Pull request by nabilatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/pulls) open
- [x] I have verified that I am following the COMET-WEB [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/COMET-WEB-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description

Fix #351 

- IsOnEditMode is now related to the Component
- The Input mode is set on contained component
- The CompoundParameterEditor reacts to the Switch changes
- Scales appears inside CompoundParameterEditor

### CODE REVIEW

File: COMET.Web.Common/ViewModels/Components/ParameterEditors/ParameterTypeEditorBaseViewModel.cs [hunk 1]
```diff
@@ -75,6 +87,12 @@ protected ParameterTypeEditorBaseViewModel(T parameterType, IValueSet valueSet,
                 this.Parameter = valueSetBase.Container as ParameterOrOverrideBase;
             }
 
+            if (isFromCompoundParameterType)
+            {
+                IsFromCompoundParameterType = true;
```
Comment (antoineatrhea) [inline COMET.Web.Common/ViewModels/Components/ParameterEditors/ParameterTypeEditorBaseViewModel.cs:92]: Missing this

File: COMET.Web.Common/Components/ParameterTypeEditors/QuantityKindParameterTypeEditor.razor [hunk 2]
```diff
@@ -30,9 +30,9 @@
                BindValueMode="@this.BindValueMode">
     </DxTextBox>
     
-    @if (this.ViewModel.Parameter.Scale != null)
+    @if (this.ViewModel.ParameterType.DefaultScale != null)
     {
-        <span class="content-scale">[@this.ViewModel.Parameter.Scale.ShortName]</span>
+        <span class="content-scale">[@this.ViewModel.ParameterType.DefaultScale.ShortName]</span>
```
Comment (antoineatrhea) [inline COMET.Web.Common/Components/ParameterTypeEditors/QuantityKindParameterTypeEditor.razor:35]: The current scale may not be the Default one. Needs to be the Parameter.Scale property. When it's inside a compoundParameterType, make sure to correctly pass the ParameterTypeComponent

File: COMET.Web.Common/ViewModels/Components/ParameterEditors/ParameterTypeEditorBaseViewModel.cs [hunk 3]
```diff
@@ -58,13 +58,25 @@ public abstract class ParameterTypeEditorBaseViewModel<T> : DisposableObject, IP
...
-        /// <param name="parameterType">the parameter type of this view model</param>
-        /// <param name="valueSet">the value set asociated to this editor</param>
-        /// <param name="isReadOnly">The readonly state</param>
-        /// <param name="valueArrayIndex">the index of the value changed in the value sets</param>
-        protected ParameterTypeEditorBaseViewModel(T parameterType, IValueSet valueSet, bool isReadOnly, int valueArrayIndex = 0)
+        /// Gets the <see cref="ParameterSwitchKind" /> of the <see cref="CompoundParameterType"/>
+        /// </summary>
+        public ParameterSwitchKind CompoundCurrentParameterSwitchKind { get; set; }
+
+		/// <summary>
```
Comment (antoineatrhea) [inline COMET.Web.Common/ViewModels/Components/ParameterEditors/ParameterTypeEditorBaseViewModel.cs:65]: Weird indentation here. Make sure that "Spaces"  is selected inside VS for this file

File: COMET.Web.Common/ViewModels/Components/ParameterEditors/ParameterTypeEditorSelectorViewModel.cs [hunk 4]
```diff
@@ -90,27 +90,55 @@ private void InitializesProperties(bool readOnly)
...
+        /// Gets the <see cref="ParameterSwitchKind" /> of the <see cref="CompoundParameterType"/>
         /// </summary>
-        /// <typeparam name="T">the parameter type</typeparam>
-        /// <returns>the view model</returns>
-        public IParameterEditorBaseViewModel<T> CreateParameterEditorViewModel<T>() where T : ParameterType
+        public ParameterSwitchKind CompoundCurrentParameterSwitchKind { get; set; }
+
+		/// <summary>
+		/// value indicating if the <see cref="ParameterSwitchKind"/> is from <see cref="CompoundParameterType"/>
+		/// </summary>
```
Comment (antoineatrhea) [inline COMET.Web.Common/ViewModels/Components/ParameterEditors/ParameterTypeEditorSelectorViewModel.cs:99]: Weird indentation also

File: COMET.Web.Common/ViewModels/Components/ParameterEditors/ParameterTypeEditorSelectorViewModel.cs [hunk 5]
```diff
@@ -90,27 +90,55 @@ private void InitializesProperties(bool readOnly)
...
+
+		/// <summary>
+		/// Creates a view model for the corresponding editor
+		/// </summary>
+		/// <typeparam name="T">the parameter type</typeparam>
+		/// <returns>the view model</returns>
+		public IParameterEditorBaseViewModel<T> CreateParameterEditorViewModel<T>() where T : ParameterType
         {
-            this.haveValueSetViewModel = this.ParameterType switch
+            if (IsFromCompoundParameterType)
```
Comment (antoineatrhea) [inline COMET.Web.Common/ViewModels/Components/ParameterEditors/ParameterTypeEditorSelectorViewModel.cs:109]: missing this

**Review by antoineatstariongroup — COMMENTED** [review]
Some indentation problem but the missing scale should be fixed (cfr. my comment)

**Review by antoineatstariongroup — APPROVED** [review]

**Comment by antoineatrhea** [comment]
@nabilatrhea Verify if this PR is still relevant

**Comment by antoineatrhea** [comment]
> @antoineatrhea This PR is not relevant anymore, see the #385 which is not handled
@nabilatrhea You can then close this PR and work on not handled feature

**Pull request closed without merging**